src/pudl/analysis/mcoe.py

```diff
@@ -400,23 +398,29 @@ def mcoe(
     the range of times and at the time resolution of the input pudl_out object.
 
     Args:
-        pudl_out (pudl.output.pudltable.PudlTabl): a PUDL output object
+        pudl_out (pudl.output.pudltabl.PudlTabl): a PUDL output object
```

Comment:
I moved all of the type hints to the call signature in response to @zaneselvans  [comment ](https://github.com/catalyst-cooperative/pudl/pull/1550#discussion_r883871312). However moving this to the call signature caused a circular import issue, so this one I left in the docstring. Not sure if it's better to have consistency and just leave out the type hint for `pudl_out`.

Reply:
I've also had the same circular reference issue :-( i don't understand what the correct implementation is. regardless, I think this is fine solution even if its not consistent